tests/integration/onchain_ops.go

```diff
@@ -0,0 +1,165 @@
+package integration
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	contracts "github.com/smartcontractkit/ccip-owner-contracts/gethwrappers"
+	storage_contract "github.com/smartcontractkit/chainlink-evm-testing/framework/contract/bind/Storage"
+	"github.com/stretchr/testify/require"
+)
+
+func DeployTimelock(
+	t *testing.T, ctx context.Context, transactor *bind.TransactOpts, backend Backend,
+	adminAccount common.Address, minDelay *big.Int,
+) (
+	common.Address, *types.Transaction, *types.Receipt, *contracts.RBACTimelock,
+) {
+	t.Helper()
+
+	proposers := []common.Address{adminAccount}
+	executors := []common.Address{adminAccount}
+	cancellers := []common.Address{adminAccount}
+	bypassers := []common.Address{}
+
+	address, transaction, contract, err := contracts.DeployRBACTimelock(
+		transactor, backend, minDelay, adminAccount, proposers, executors, cancellers, bypassers)
+	require.NoError(t, err)
+
+	backend.Commit()
```

Comment:
we need to explicitly call `Commit` before `WaitMined` for the in-memory simulated backend to work